### PR TITLE
expose tracing config in IngressRoute

### PR DESCRIPTION
**Pull request by phylake**

### CODE REVIEW

File: internal/contour/route.go [hunk 1]
```diff
@@ -143,8 +145,22 @@ func (v *routeVisitor) visit(vertex dag.Vertex) {
 							rt.ResponseHeadersToAdd = envoy.HeaderValueList(route.ResponseHeadersPolicy.Set, false)
 							rt.ResponseHeadersToRemove = route.ResponseHeadersPolicy.Remove
 						}
-						routes = append(routes, rt)
 					}
+
+					if route.Tracing != nil {
+						rt.Tracing = &envoy_api_v2_route.Tracing{
+							ClientSampling: &envoy_type.FractionalPercent{
+								Numerator:   uint32(route.Tracing.ClientSampling),
```
Comment (lrouquette) [inline internal/contour/route.go:153]: We talked about implementing an optional override via env var; did you deem that no longer necessary?

File: internal/envoy/route.go [hunk 2]
```diff
@@ -57,10 +57,9 @@ func RouteMatch(route *dag.Route) *envoy_api_v2_route.RouteMatch {
 // weighted cluster.
 func RouteRoute(r *dag.Route) *envoy_api_v2_route.Route_Route {
 	ra := envoy_api_v2_route.RouteAction{
-		Timeout:               r.Timeout,
-		IdleTimeout:           r.IdleTimeout,
-		PrefixRewrite:         r.PrefixRewrite,
-		RequestMirrorPolicies: mirrorPolicy(r),
```
Comment (lrouquette) [inline internal/envoy/route.go:63]: ? conflict perhaps?

Comment (phylake) [reply internal/envoy/route.go:63]: No i removed it. Not as big of a deal now that we have `preserveUnknownFields: false` but old habits die hard

File: internal/contour/route.go (hunk 1 above)
Comment (phylake) [reply internal/contour/route.go:153]: Forgot about that. Now it's the same env var that listener_adobe.go uses to enable tracing on the HTTPConnectionManager

**Pull request merged**